packaging/firewall/alvr_fw_config.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+# Basic script to add / remove firewall configuration for ALVR
+# Usage: ./alvr_fw_config.sh add|remove
+# Exit codes:
+# 1 - Invalid command
+# 2 - Invalid action
+# 99 - Feature not implemented
+# 126 - pkexec failed - Request dismissed
+
+iptables_cfg() {
+    exit 99
+}
+
+firewalld_cfg() {
+    # Iterate around each active zone
+    for zone in $(firewall-cmd --get-active-zones | grep -P '^\w+.*\w$'); do
+        if [ "${1}" == 'add' ]; then
+            # If running or permanent alvr service is missing, add it
+            if ! firewall-cmd --zone="${zone}" --list-services | grep 'alvr' >/dev/null 2>&1; then
+                firewall-cmd --zone="${zone}"  --add-service='alvr'
+            fi
+            if ! firewall-cmd --zone="${zone}" --list-services --permanent | grep 'alvr' >/dev/null 2>&1; then
+                firewall-cmd --zone="${zone}"  --add-service='alvr' --permanent
+            fi
+        elif [ "${1}" == 'remove' ]; then
+            # If running or persistent alvr service exists, remove it
+            if firewall-cmd --zone="${zone}" --list-services | grep 'alvr' >/dev/null 2>&1; then
+                firewall-cmd --zone="${zone}"  --remove-service='alvr'
+            fi
+            if firewall-cmd --zone="${zone}" --list-services --permanent | grep 'alvr' >/dev/null 2>&1; then
+                firewall-cmd --zone="${zone}"  --remove-service='alvr' --permanent
+            fi
+        else
+            exit 2
+        fi
+    done
+}
+
+ufw_cfg() {
+    # Try and install the application file
+    if ! ufw app info 'alvr'; then
+        if [ -f '/usr/share/alvr/ufw-alvr' ]; then
+            cp '/usr/share/alvr/ufw-alvr' '/etc/ufw/applications.d/'
+        elif [ -f './ufw-alvr' ]; then
+            cp './ufw-alvr' '/etc/ufw/applications.d/'
+        else
+            exit 3
+        fi
+    fi
+
+    if [ "${1}" == 'add' ] && ! ufw status | grep 'alvr' >/dev/null 2>&1; then
+        ufw allow 'alvr'
+    elif [ "${1}" == 'remove' ] && ufw status | grep 'alvr' >/dev/null 2>&1; then
+        ufw delete allow 'alvr'
+    else
+        exit 2
+    fi
+}
+
+main() {
+    # If we're not root use pkexec for GUI prompt
+    if [ "$(whoami)" == 'root' ]; then
```

Comment:
```suggestion
    if [ "$USER" == 'root' ]; then
```

Reply:
The use of `$(whoami)` is intentional; using `$USER` will not necessarily return root if the script is run with `sudo` in some ways, ex:
```
sh-5.1$ whoami
trae
sh-5.1$ echo "${USER}"
trae
sh-5.1$ sudo whoami
root
sh-5.1$ sudo echo "${USER}"
trae
```